src/marin/run/ray_run.py

```diff
@@ -19,82 +19,41 @@
 import os
 import re
 import shlex
+import time
+from urllib.parse import urlparse
 
-import toml
 from ray.job_submission import JobSubmissionClient
 
-from marin.run.vars import ENV_VARS, PIP_DEPS, REMOTE_DASHBOARD_URL
+from marin.run.vars import REMOTE_DASHBOARD_URL
 
-# Setup logger
-logger = logging.getLogger("ray")
+logger = logging.getLogger(__name__)
 
 
-def parse_pip_requirements(line: str) -> list[str]:
-    # The pattern means:
-    #   (?:\[[^\]]*\]|[^,])+
-    #   Match either:
-    #       - a bracketed chunk: [anything but ]]
-    #       - OR any character that isn't a comma
-    #   Repeat 1 or more times.
-    # So each match is all characters up to the next top-level comma.
-    # For example, the input string "numpy==2.0.0,scipy[extras1,extras2],sympy"
-    # is parsed as ["numpy==2.0.0", "scipy[extras1,extras2]", "sympy"]
-    pattern = r"(?:\[[^\]]*\]|[^,])+"
-    return re.findall(pattern, line)
+def parse_user_command_line(command: str) -> dict[str, str | None]:
+    """Parse command line to extract experiment name and prefix for submission ID."""
+    experiment = re.search(r"experiments/([^ ]+)", command)
 
+    try:
+        if experiment:
+            experiment = experiment.group(1).split("/")[-1].split(".")[0]
+    except Exception:
+        experiment = ""
 
-def generate_pythonpath(base_dir="submodules"):
```

Comment:
so we do want this submoduel thing still. it's how we do codevelopment of levanter and marin